### PR TITLE
Interval feature for TOTP

### PR DESCRIPTION
**Pull request by pedrofurtado**
@robertomiranda You review on it will be awesome 🤝

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,6 +213,23 @@ user.provisioning_uri(nil, issuer: 'MYAPP') #=> 'otpauth://totp/hello@heapsource
 
 This can then be rendered as a QR Code which can be scanned and added to the users list of OTP credentials.
 
+If you define a custom interval for TOTP codes, just as `has_one_time_password interval: 10` (for example), remember to include the interval also in `provisioning_uri` method.
```
[the remark this answers is not in the document]
Comment (robertomiranda) [reply README.md:216]: @pedrofurtado could we clarify that the interval given is in seconds and the default value is 30? I think would be a common question from this feature

https://github.com/mdp/rotp/blob/62874be71d74380d252c73409dd81da08b021497/lib/rotp/totp.rb#L2-L12

File: README.md [hunk 2]
```diff
@@ -213,6 +213,23 @@ user.provisioning_uri(nil, issuer: 'MYAPP') #=> 'otpauth://totp/hello@heapsource
...
+  has_one_time_password interval: 10
+end
+
+user = User.new
+user.provisioning_uri("hello", interval: 10) # => 'otpauth://totp/hello?secret=2z6hxkdwi3uvrnpn&period=10'
+```
+
+> Note that only some authenticator apps are compatible with custom `period` of tokens, for more details check these links:
+> https://labanskoller.se/blog/2019/07/11/many-common-mobile-authenticator-apps-accept-qr-codes-for-modes-they-dont-support/
+> https://www.ibm.com/docs/en/sva/9.0.7?topic=authentication-configuring-totp-one-time-password-mechanism
```
Comment (robertomiranda) [inline README.md:229]:
```suggestion
**Note**: that only some authenticator apps are compatible with custom `period` of tokens, for more details check these links:

- https://labanskoller.se/blog/2019/07/11/many-common-mobile-authenticator-apps-accept-qr-codes-for-modes-they-dont-support/
- https://www.ibm.com/docs/en/sva/9.0.7?topic=authentication-configuring-totp-one-time-password-mechanism
```

File: lib/active_model/one_time_password.rb [hunk 3]
```diff
@@ -7,13 +7,15 @@ def has_one_time_password(options = {})
...
-                        :otp_backup_codes_count, :otp_one_time_backup_codes
+                        :otp_backup_codes_count, :otp_one_time_backup_codes,
+                        :otp_interval
 
         self.otp_column_name = (options[:column_name] || "otp_secret_key").to_s
         self.otp_digits = options[:length] || 6
 
         self.otp_counter_based = (options[:counter_based] || false)
         self.otp_counter_column_name = (options[:counter_column_name] || "otp_counter").to_s
+        self.otp_interval = options[:interval] || 30
```
Comment (robertomiranda) [inline lib/active_model/one_time_password.rb:18]: Can we trust the default value is assigned by TOTP constructure https://github.com/mdp/rotp/blob/master/lib/rotp/totp.rb#L9?

```suggestion
        self.otp_interval = options[:interval]
```

If not what about moving this into a constant?

File: lib/active_model/one_time_password.rb [hunk 4]
```diff
@@ -7,13 +7,15 @@ def has_one_time_password(options = {})
...
         class_attribute :otp_digits, :otp_counter_based,
-                        :otp_backup_codes_count, :otp_one_time_backup_codes
+                        :otp_backup_codes_count, :otp_one_time_backup_codes,
+                        :otp_interval
 
         self.otp_column_name = (options[:column_name] || "otp_secret_key").to_s
         self.otp_digits = options[:length] || 6
 
         self.otp_counter_based = (options[:counter_based] || false)
         self.otp_counter_column_name = (options[:counter_column_name] || "otp_counter").to_s
```
Comment (robertomiranda) [inline lib/active_model/one_time_password.rb:17]: Not part of this PR but would be good to start to move the default values to class constants

**Review by robertomiranda — COMMENTED** [review]
Looks great @pedrofurtado just minor suggestions around the notes in the README file and a nitpick around how we handle the default values that would be good to address before merging this PR 👍 

Thank you 🙇

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (pedrofurtado) [reply README.md:216]: Good idea 💡

File: lib/active_model/one_time_password.rb (hunk 4 above)
Comment (pedrofurtado) [reply lib/active_model/one_time_password.rb:17]: I created another PR for this suggestion 🎉 https://github.com/heapsource/active_model_otp/pull/89

**Comment by pedrofurtado** [comment]
@robertomiranda Suggestions applied! Let me know your thoughts about the changes 👍

**Review by robertomiranda — APPROVED** [review]

**Comment by robertomiranda** [comment]
Nice one @pedrofurtado! 👏
